Userland/Libraries/LibJS/Runtime/GlobalObject.cpp

```diff
@@ -110,18 +113,25 @@ void GlobalObject::initialize_global_object()
     // This must be initialized before allocating AggregateErrorPrototype, which uses ErrorPrototype as its prototype.
     m_error_prototype = heap().allocate<ErrorPrototype>(*this, *this);
 
-#define __JS_ENUMERATE(ClassName, snake_name, PrototypeName, ConstructorName, ArrayType) \
-    if (!m_##snake_name##_prototype)                                                     \
-        m_##snake_name##_prototype = heap().allocate<PrototypeName>(*this, *this);
-    JS_ENUMERATE_BUILTIN_TYPES
-#undef __JS_ENUMERATE
-
 #define __JS_ENUMERATE(ClassName, snake_name) \
     if (!m_##snake_name##_prototype)          \
         m_##snake_name##_prototype = heap().allocate<ClassName##Prototype>(*this, *this);
     JS_ENUMERATE_ITERATOR_PROTOTYPES
 #undef __JS_ENUMERATE
 
+    // %GeneratorFunction.prototype.prototype% must be initialized separately as it has no
+    // companion constructor
+    m_generator_object_prototype = heap().allocate<GeneratorObjectPrototype>(*this, *this);
+    if (vm.exception())
+        return;
```

Comment:
Why would this throw?

Reply:
Pretty sure I meant to have this under the `define_property` line, but realistically that shouldn't throw either. `initialize_constructor` has that check though, which is the reason I kept it around.

Reply:
I see - `initialize_constructor` shouldn't need either of the checks it has :smile:
Let's remove this one for now and deal with `initialize_constructor` separately.